src/main/java/com/avairebot/imagegen/RankBackgroundHandler.java

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright (c) 2018.
+ *
+ * This file is part of AvaIre.
+ *
+ * AvaIre is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * AvaIre is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with AvaIre.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ *
+ */
+
+package com.avairebot.imagegen;
+
+import com.avairebot.Constants;
+import com.avairebot.shared.ExitCodes;
+import com.avairebot.utilities.ResourceLoaderUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.util.*;
+
+public class RankBackgroundHandler {
+
+    private static final Logger log = LoggerFactory.getLogger(RankBackgroundHandler.class);
+    private static final LinkedHashMap<RankBackground, BackgroundRankColors> backgroundColors = new LinkedHashMap<>();
+    private static final LinkedHashMap<String, Integer> namesToCost = new LinkedHashMap<>();
+    private static final List<RankBackground> backgrounds = new ArrayList<>();
+    private static final List<Integer> usedIds = new ArrayList<>();
+    private static RankBackgroundHandler instance;
+    private static File backgroundsFolder;
+    private static boolean backgroundsFolderAlreadyExists = false;
```

Comment:
All of these methods are static, and most of them have a public static method that can be used to access them directly, if the class is suppose to be a singleton, the only static property should be the instance variable.